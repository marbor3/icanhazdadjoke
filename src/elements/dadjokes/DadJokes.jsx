import react, { useEffect, useState } from "react";
import DadJoke from "../dadjoke/DadJoke";
import styles from "./DadJokes.module.css";

export default function DadJokes({ jokes }) {
  return jokes && jokes.length ? (
    <ul className={styles.jokes}>
      {jokes.map((joke) => (
        <li
          data-testid="joke-list-item"
          key={joke.id}
          className={styles["jokes__list-item"]}
        >
          <DadJoke joke={joke.joke} />
        </li>
      ))}
    </ul>
  ) : null;
}
