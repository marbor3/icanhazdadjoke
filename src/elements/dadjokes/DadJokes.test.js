import React from "react";
import { render } from "@testing-library/react";
import DadJokes from "./DadJokes";

describe("Elements - DadJokes", () => {
  test("without any jokes theres nothing to render", async () => {
    const { queryByTestId, container } = render(<DadJokes />);

    expect(queryByTestId("joke-list-item")).not.toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test("with empty jokes list theres nothing to render", async () => {
    const { queryByTestId, container } = render(<DadJokes jokes={[]} />);

    expect(queryByTestId("joke-list-item")).not.toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test("given an array of jokes, it renders all jokes", async () => {
    const jokes = [
      {
        id: "0ga2EdN7prc",
        joke: "How come the stadium got hot after the game? Because all of the fans left.",
      },
      {
        id: "0oO71TSv4Ed",
        joke: "Why was it called the dark ages? Because of all the knights. ",
      },
      { id: "0oz51ozk3ob", joke: "A steak pun is a rare medium well done." },
    ];
    const { getAllByTestId, container } = render(<DadJokes jokes={jokes} />);

    expect(getAllByTestId("joke-list-item")).toHaveLength(3);
    expect(container).toMatchSnapshot();
  });
});
