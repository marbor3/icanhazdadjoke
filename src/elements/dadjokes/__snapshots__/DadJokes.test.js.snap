// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Elements - DadJokes given an array of jokes, it renders all jokes 1`] = `
<div>
  <ul
    class="jokes"
  >
    <li
      class="jokes__list-item"
      data-testid="joke-list-item"
    >
      <article
        class="article"
        data-testid="joke"
      >
        How come the stadium got hot after the game? Because all of the fans left.
      </article>
    </li>
    <li
      class="jokes__list-item"
      data-testid="joke-list-item"
    >
      <article
        class="article"
        data-testid="joke"
      >
        Why was it called the dark ages? Because of all the knights. 
      </article>
    </li>
    <li
      class="jokes__list-item"
      data-testid="joke-list-item"
    >
      <article
        class="article"
        data-testid="joke"
      >
        A steak pun is a rare medium well done.
      </article>
    </li>
  </ul>
</div>
`;

exports[`Elements - DadJokes with empty jokes list theres nothing to render 1`] = `<div />`;

exports[`Elements - DadJokes without any jokes theres nothing to render 1`] = `<div />`;
